napalm_eos/eos.py

```diff
@@ -1277,7 +1277,7 @@ def _parse_per_peer_bgp_detail(peer_output):
                         unicode, item['last_event']))
                 item['remote_address'] = (
                     napalm_base.helpers.convert(
-                        unicode, item['remote_address']))
+                        napalm_base.helpers.ip, item['remote_address']))
```

Comment:
convert() looks to call the first argument (napalm_base.helpers.ip) with the second argument (item['remote_address']):

```
def convert(to, who, default=u''):
    if who is None:
        return default
    try:
        return to(who)
    except:
        return default
```

So I don't think it's appropriate to pass napalm_base.helpers.ip(item['remote_address'])).

You could argue that the whole convert part is unnecessary, but I went for minimal changes.